sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -227,3 +228,14 @@ object ReuseSubquery extends Rule[SparkPlan] {
     }
   }
 }
+
+object PrepareScans extends Rule[SparkPlan] {
+  def apply(plan: SparkPlan): SparkPlan = {
+    val scans = plan.collect {
+      case scan: BatchScanExec => scan
+    }
+    scans.foreach(_.prepare())
```

Comment:
I mention in the doc why I am using `prepare` but we make this more specific to dynamic filters if needed.

Reply:
Thank you for adding this rule. BTW, if possible, please make a new file for this rule.

Reply:
Sure! Will move to a new file.

Reply:
Resolving this comment as this rule does not exist anymore.